File: C/MatMultiply.c
//Matrix multiplication using function

//Function to print Matrices.
#include <stdio.h>
//Function to Print Matrix
int PrintMatrix(int print[10][10], int rows, int columns){
    int i, j;
    for(i=0; i<rows; i++){
        for(j=0; j<columns; j++){
        printf("\t%d", print[i][j]);
        }
    printf("\n");
    }
}

//Function to Multiply Matrices.
int MultiplyMatrix(int matrix1[10][10], int matrix2[10][10], int Multiplied[10][10], int row1, int column1, int row2, int column2){
    int i, j, k;
    for(i=0; i<row1; i++){
        for(j=0; j<column1 ; j++){
            Multiplied[i][j]=0;
            for(k=0; k<column1 ; k++){
                Multiplied[i][j]=Multiplied[i][j]+matrix1[i][k]*matrix2[k][j];
            }
        }
    }
}

int main(){
int matrix1[10][10], matrix2[10][10], Multiplied[10][10];
int i, j, row1, column1, row2, column2;
printf("Enter no. of rows & columns of Matrix 1: "); //Entering no. of Rows & Columns of 1st Matrix
scanf("%d %d", &row1, &column1);
printf("Enter no. of rows & columns of Matrix 2: "); //Entering no. of Columns & Rows of 1st Matrix
scanf("%d %d", &row2, &column2);

if(row1==column2)
    printf("\n\nMatrix can be Multiplied.");
else
    printf("\n\nMatrix cannot be Multiplied. As Row of 1st Matrix not equal to Column of 2nd Matrix");

            //Simultaneously getting values of 1st & 2nd Matrices.
            printf("\n\n\nEnter value of 1st Matrix\n");

            for(i=0; i<row1; i++){
            for(j=0; j<column1; j++){
            printf("\nEnter Matrix1[%d][%d] value: ", i, j);
            scanf("%d", &matrix1[i][j]);
                                    }
                                 }
            printf("\n\n\nEnter value of 2nd Matrix\n");     
            for(i=0; i<row2; i++){
            for(j=0; j<column2; j++){
            printf("\nEnter Matrix2[%d][%d] value: ", i, j);
            scanf("%d", &matrix2[i][j]);
                                    }
            }
                                    
    //Printing Product of Matrix 1 & 2.
    printf("\nProduct of Matrices\n");
    MultiplyMatrix(matrix1, matrix2, Multiplied, row1, column1, row2, column2);
    PrintMatrix(Multiplied, row1, column2);

    return 0;
}

File: Java/MatMultiply.java
//Java program to multiply two matrices.

import java.util.*;
class MatMultiply{
    //Method to get Matrix1.
    static int[][] GetMatrix1(int rows, int columns){
    Scanner scan = new Scanner(System.in); //new input stream for GetMatrix Method
    int[][] matrix1 = new int[10][10];
    int i, j;
    for(i=0; i<rows; i++){
    for(j=0; j<columns; j++){
        System.out.print("Enter a["+(i+1)+"]["+(j+1)+"] value: ");
        matrix1[i][j] = scan.nextInt();
    }
}
    return matrix1;
}
    //Method to get Matrix2.
    static int[][] GetMatrix2(int rows, int columns){
        Scanner scan = new Scanner(System.in); //new input stream for GetMatrix2 Method
        int[][] matrix2 = new int[10][10];
        int i, j;
        for(i=0; i<rows; i++){
        for(j=0; j<columns; j++){
            System.out.print("Enter b["+(i+1)+"]["+(j+1)+"] value: ");
            matrix2[i][j] = scan.nextInt();
        }
    }
        scan.close();
        return matrix2;
    }
    
    //Method to print Matrix.
    static void PrintMatrix(int[][] prints, int rows, int columns){
    int i, j;
    for(i=0; i<rows; i++){
        for(j=0; j<columns; j++){
        System.out.print(prints[i][j]+"\t");
        }
    System.out.print("\n");
    }
}
    
    //Method to Multiply Matrices
    static void MultiplyMat(int[][] matrix1, int[][] matrix2, int[][] matrix3, int row1, int column1, int column2){
    	int i, j, k;
    	for( i= 0 ; i < row1 ; i++ ) {
			for ( j= 0 ; j < column2;j++){
    		     for ( k= 0 ; k < column1; k++ ){
    		    	 matrix3[i][j] = matrix3[i][j] + (matrix1[i][k] * matrix2[k][j]);
    		           }
    		     System.out.print(matrix3[i][j] + " ");
    		}
			System.out.println();
 	}
}
    
     //main method
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in); //new input stream
        int row1, column1, row2, column2;

        System.out.print("Enter no. of rows & column in Matrix1: "); //Getting no. of Rows
        row1 = scan.nextInt();
        column1 = scan.nextInt();
        System.out.print("Enter no. of rows & columns in Matrix2: "); //Getting no. of Columns
        row2 = scan.nextInt();
        column2 = scan.nextInt();
        
        scan.close();
    
        //Checking whether matrices can be multiplied or not (column1 = row2).
        if(column1 == row2) {
        	System.out.print("Matrix can be multiplied\n");
            
            //Getting values of Matrix1 & Matrix2.
            int[][] matrix1 = GetMatrix1(row1, column1);
            int[][] matrix2 = GetMatrix2(row2, column2);
            int matrix3[][] = new int[row1][column2]; //initializing matrix3
            
            //Printing Matrix1 & Matrix2 
            System.out.print("\t\t\nInput Matrix1:\n");
            PrintMatrix(matrix1, row1, column1);
            System.out.print("\t\t\nInput Matrix2:\n");
            PrintMatrix(matrix2, row2, column2);
            
            //Multiplying two matrices & printing the resultant
            System.out.print("\t\t\nResultant Matrix:\n");
            MultiplyMat(matrix1, matrix2, matrix3, row1, row2, column2);
    }
        else
        	System.out.print("Matrix cannot be multiplied\n");
    }
    
}    